tests/util/golinter/README.md

```diff
@@ -0,0 +1,53 @@
+# golinter
+
+This golinter checks if our test files follow customerized lint rules.
+
+# Unit tests rules.
+## File name requirement.
+Unit tests should stay in file with suffix "_test.go".
+## Forbidden function calls.
+Unit tests should not contain any of these function calls.
+```bash
+testing.Short()
+time.Sleep()
+goroutine
+```
+
+# Integration tests
+## File name requirement
+Integration tests should stay in file with suffix "_integ.go" in any directory, or file with suffix "_integ.go"/"_test.go" but under a directory named "integ".
```

Comment:
Anyone else using this convention ? If yes, please add a comment. 
Did we find any other golang project using similar patterns ? And is it needed - if tests are annotated, and same test could run in multimple env - why would we use different files ?

Reply:
@costinm "why would we use different files?" I am not sure what you mean here.

The convention we are proposing should be non intrusive for most tests files.
i.e. unit tests with _test.go suffix can exist in any packages. e2e tests are in dedicated packages.

The only questionable ones are the integration tests that look like unit tests which may not always run out of the box. And that is exactly what we are trying to address here.